src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/RoutePatternAnalyzer.cs

```diff
@@ -34,22 +34,39 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
         var routeUsageCache = RouteUsageCache.GetOrCreate(context.SemanticModel.Compilation);
 
         // Update to use FilterSpan when available. See https://github.com/dotnet/aspnetcore/issues/48157
-        foreach (var item in root.DescendantTokens())
```

Comment:
>foreach (var item in root.DescendantTokens()) [](http://example.com/codeflow?start=8&length=45)

Was this change needed to fix the issue? The reason for asking is if the fix may be considered for .NET 8, it might be useful to reduce the size of the change.